packages/lib/src/interactions/SelectionTool.tsx

```diff
@@ -131,10 +129,30 @@ function SelectionTool(props: Props) {
   );
 
   useEffect(() => {
-    if (selection && isModifierKeyPressed) {
-      onSelectionChangeRef.current?.(selection);
+    if (!prevSelection && selection) {
+      onSelectionStartRef.current?.(selection);
+      return;
+    }
+
+    if (prevSelection && !selection && hasSuccessfullyEndedRef.current) {
+      hasSuccessfullyEndedRef.current = false;
+      onSelectionEndRef.current?.(prevSelection);
+      return;
+    }
+
+    if (prevSelection && selection) {
+      onSelectionChangeRef.current?.(
+        isModifierKeyPressed ? selection : undefined
```

Comment:
Could you elaborate a bit on the reason of this change ?

Reply:
Ah yeah, good point. Basically, this replicates the behaviour of the `children` render prop: we didn't use to inform `onSelectionChange` subscribers when the selection was no longer "valid"/"active"/"relevant" (whatever you want to call it), when the modifier key was released. The callback stopped being called with no other cues whatsoever. Now, they get `undefined` once when the modifier key is released, and then the latest `selection` object when the modifier key is pressed again.

Reply:
I see. Clever!

That would need a bit of documentation I think though.

Reply:
User-facing? It is demonstrated in the Storybook, technically. Or do you mean dev-facing documentation? I'll add a comment.